File: sources/net.sf.j2s.core/test/src/com/falstad/circuit/ImportExportDialogSwingJS.js
(function(){var P$=Clazz.newPackage("com.falstad.circuit"),I$=[];
var C$=Clazz.newClass(P$, "ImportExportDialogSwingJS", null, 'a2s.Dialog', 'com.falstad.circuit.ImportExportDialog');
C$.lastName = null;

C$.$clinit$ = function() {Clazz.load(C$, 1);
C$.lastName = "circuit.txt";
}

Clazz.newMeth(C$, '$init0$', function () {
var c;if((c = C$.superclazz) && (c = c.$init0$))c.apply(this);
this.cframe = null;
this.text = null;
this.$type = null;
}, 1);

Clazz.newMeth(C$, '$init$', function () {
}, 1);

Clazz.newMeth(C$, 'c$$com_falstad_circuit_CirSim$com_falstad_circuit_ImportExportDialog_Action', function (f, type) {
C$.superclazz.c$$a2s_Frame$S$Z.apply(this, [f, (type === (I$[0]||(I$[0]=Clazz.load(Clazz.load('com.falstad.circuit.ImportExportDialog').Action))).EXPORT ) ? "Export" : "Import", false]);
C$.$init$.apply(this);
this.cframe = f;
this.$type = type;
this.text = Clazz.new_((I$[1]||(I$[1]=Clazz.load('a2s.TextArea'))).c$$S$I$I,["", 10, 60]);
}, 1);

Clazz.newMeth(C$, 'setDump$S', function (dump) {
this.text.setText$S(dump);
});

Clazz.newMeth(C$, 'execute', function () {
if (this.$type === (I$[0]||(I$[0]=Clazz.load(Clazz.load('com.falstad.circuit.ImportExportDialog').Action))).IMPORT ) {
{
swingjs.JSToolkit.getFileFromDialog(this, "string");
}} else {
var data = this.text.getText();
var mimeType = "text/plain";
var encoding = null;
var fileName = null;
var name = C$.lastName;
{
fileName = prompt("Enter a file name", name); fileName && swingjs.JSToolkit.saveFile(fileName, data, mimeType, encoding);
}}this.dispose();
return;
});

Clazz.newMeth(C$, 'handleFileLoaded$O$S', function (data, fileName) {
if (fileName == null ) return;
C$.lastName = fileName;
try {
this.cframe.readSetup$S(data);
} finally {
this.dispose();
}
});

Clazz.newMeth(C$, 'handleEvent$java_awt_Event', function (ev) {
if (ev.id == 201) {
this.cframe.main.requestFocus();
this.setVisible$Z(false);
this.cframe.impDialog = null;
return true;
}return C$.superclazz.prototype.handleEvent$java_awt_Event.apply(this, [ev]);
});

Clazz.newMeth(C$);
})();
//Created 2017-12-17 19:28:19

File: sources/net.sf.j2s.java.core/src2/com/falstad/circuit/ImportExportDialogSwingJS.java
package com.falstad.circuit;

import java.awt.Event;

import java.awt.Dialog;
import java.awt.TextArea;

/**
 * 
 * @author Bob Hanson
 * 
 */
class ImportExportDialogSwingJS extends Dialog implements ImportExportDialog {
	private CirSim cframe;
	private TextArea text;
	private Action type;

	ImportExportDialogSwingJS(CirSim f, Action type) {
		super(f, (type == Action.EXPORT) ? "Export" : "Import", false);
		cframe = f;
		this.type = type;
		text = new TextArea("", 10, 60);
		// we don't really create this dialog for a SwingJS import or export
	}

	@Override
	public void setDump(String dump) {
		text.setText(dump);
	}

	private static String lastName = "circuit.txt";

	@SuppressWarnings("unused")
	@Override
	public void execute() {
		if (type == Action.IMPORT) {
			/**
			 * @j2sNative
			 * 
			 *            swingjs.JSToolkit.getFileFromDialog(this, "string");
			 */
			{
			}
		} else {
			String data = text.getText();
			String mimeType = "text/plain";
			String encoding = null;
			String fileName = null;
			String name = lastName;
			/**
			 * @j2sNative
			 * 
			 *            fileName = prompt("Enter a file name", name); fileName &&
			 *            swingjs.JSToolkit.saveFile(fileName, data, mimeType,
			 *            encoding);
			 */
			{
			}
		}
		dispose();
		return;
	}

	// @Override
	public void handleFileLoaded(Object data, String fileName) {
		// SwingJS asynchronous return from FileReader
		if (fileName == null)
			return;
		lastName = fileName;
		try {
			cframe.readSetup((String) data);
		} finally {
			dispose();
		}
	}

	@Override
	public boolean handleEvent(Event ev) {
		if (ev.id == Event.WINDOW_DESTROY) {
			cframe.main.requestFocus();
			setVisible(false);
			cframe.impDialog = null;
			return true;
		}
		return super.handleEvent(ev);
	}

}
